pkg/http/api.go

```diff
@@ -900,6 +951,10 @@ func (a *api) onDeleteActorState(c *routing.Context) error {
 }
 
 func (a *api) onGetMetadata(c *routing.Context) error {
+	// Get the new context with the required trace values.
+	ctx, cancel := newContext(c)
+	defer cancel()
```

Comment:
Do we need to call cancel() always? isn't it cancelled automatically?

Reply:
> Do we need to call cancel() always? isn't it cancelled automatically?

Yes, we have to cancel manually.  I am considering to remove cancel , because context is not passed to external components. So the call to database might continue and succeed, even if we cancel the context. The only caveat will be that it will leak memory if we don't cancel the context.